client.go

```diff
@@ -18,6 +19,9 @@ import (
 // Using a client directly allows more fine-grained control over how downloading
 // is done, as well as customizing the protocols supported.
 type Client struct {
+	// ctx for cancellation
+	ctx context.Context
```

Comment:
Oh, I agree ! This would force using options when we might not always need them.
I just exported `Ctx` 🙂.

I initially did that because I thought it was safer. But this is equivalent.

Reply:
This has been addressed in bde4550. I was wondering in light of this if it made sense to keep `WithContext`, but I think it's fine in the spirit of reducing boilerplate in code making use of a context without assembling a whole client from scratch via the basic `Get` and what not functions. Plausibly, other helpers may be added to `client_option.go` at a later time as well for other common options.

Reply:
Just confirmed with @mitchellh that we're good here. Going to mark this one as resolved and merge - great job again @azr!